building-and-deploying-a-fastapi-app.md

```diff
@@ -0,0 +1,541 @@
+# Building a FastAPI Application and Deploying to Okteto
+
+
+In this tutorial, you'll learn how to develop a CRUD API with [FastAPI](https://fastapi.tiangolo.com) and deploy the application to [Okteto](https://okteto.com).
+
+You'll start by building the application, after which you will now deploy it to Okteto.
+
+## What is FastAPI
+
+FastAPI is a modern Python web framework designed for building fast and efficient backend applications. It comes with built-in support for data validation, authentication, and interactive API documentation powered by OpenAPI and Swagger.
+
+## What is Okteto?
+
+Okteto is a tool used to develop and accelerate the development workflow of the Kubernetes application.
+
+## Initial Setup
+
+Start by creating a new folder to hold your project called 'fastapi-crud':
+
+```sh
+$ mkdir fastapi-crud
+$ cd fastapi-crud
+```
+
+Next, create and activate a virtual environment in the project folder:
```

Comment:
could you explain why we need the virtualenv? (also, we should list the dependencies somewhere)

Reply:
Oh, I've added a brief reason on why we need virtualenv. I already listed that dependencies should be in the `requirements.txt` file

Reply:
perfect, thanks!